src/routes/v1/auth.js

```diff
@@ -150,17 +151,22 @@ async function whoami(req, res) {
     return res.status(200).json({ userId, email, contactNumber })
   } catch (err) {
     clearAllCookies(res)
-    // Return a 401 os that user will be redirected to logout
+    // Return a 401 so that user will be redirected to logout
     return res.sendStatus(401)
   }
 }
 
 router.get("/github-redirect", attachReadRouteHandlerWrapper(authRedirect))
-router.get("/", attachReadRouteHandlerWrapper(githubAuth))
+router.get(
+  "/",
+  statsMiddleware.trackGithubLogins,
```

Comment:
actually it's a misconception on my part - this is because we don't `await` on the promise here, so it's fine. i'm going to add a test to check that throwing errors within the promise doesn't impact app code (no error recovery cos it's only a single data point) 

eg:
```ts
router.get("/", 
  useSomeMiddleware, // throws, want to make sure this doesn't impact `handler` call
  handler
```

Reply:
okay sure, i think as long as we catch the error thrown will be fine?

Reply:
ya added a `logger.log` here in the `catch` section